Sage/carbon
# fix(decimal): allow decimal to handle dead key characters - FE-3334

This fix allows the user to use a dead key character without causing the component to unexpectedly
crash when onBlur is triggered.

fixes #3372

### Proposed behaviour

This fix allows the user to use a dead key character without causing the component to unexpectedly
crash when onBlur is triggered.

![Decimal_carrat_fix](https://user-images.githubusercontent.com/56251247/106626047-35018780-656f-11eb-9c2f-311af504f4db.gif)

### Current behaviour

If a dead key character is used and then the component is blurred, the component crashes.

![Decimal_crash_carrat](https://user-images.githubusercontent.com/56251247/106625566-ae4caa80-656e-11eb-97e7-e0f38a93b9ff.gif)

![Decimal_crash_carrat2](https://user-images.githubusercontent.com/56251247/106625813-f2d84600-656e-11eb-8512-baf90e7901c0.gif)

### Checklist


<del>[ ]  Storybook added or updated if required </del>
<del>[ ] Typescript `d.ts` file added or updated if required </del>

### Additional context
Please ensure you have your keyboard set to `German - Standard` on your Mac before testing the scenario set out in issue #3372 

### Testing instructions

- `git checkout FE-3334_decimal_fields_crashes`
- `npm i`
- `npm start` to start Storybook
- Navigate to Experimental -> Decimal -> default
- Try to replicate the scenario presented on issue #3372 *Please make sure you have your keyboard set to `German - Standard`*
- Component should not crash Storybook
